homework/index.js

```diff
@@ -30,17 +35,125 @@
     return elem;
   }
 
+  function createDetails(leftColumn, repositories) {
+    const table = createAndAppend('table', leftColumn, { id: 'main-table' });
+    const tBody = createAndAppend('tbody', table);
+    const repoDetails = ['Repository', 'Description', 'Forks', 'Updated'];
+    repoDetails.forEach(detail => {
+      const tr = createAndAppend('tr', tBody, { class: `${detail.toLowerCase()}-row` });
```

Comment:
You are using CSS classes to get a reference to them later in your code, querying the DOM with `querySelector`. While this works it would be better and more efficient to stay as much as possible in 'JavaScript land'. You could extend the approach you have chosen using a data structure (the `repoDetails` array), but a functional approach results in simpler code.

Basically, what needs to be done for each row in the table is to create a `tr` (parent - with two `td` children. The first one to hold label text and the second one to hold a property value from the repository object. So the fiunction could like this:

```js
function addRow(tbody, label, value) {
  const row = createAndAppend('tr', tbody);
  createAndAppend('td', row, { text: `${label} :`, class: 'label' });
  createAndAppend('td', row, { text: value });
  return row;
}
```

Now it is easy to add a row, e.g.:

```js
addRow(tbody, 'Forks', repo.forks);
```

For the first row, you would need to do something special, as it does not use simple text but an `<a>` tag:

```js
const firstRow = addRow(tbody, 'Repository');
createAndAppend('a', firstRow.lastChild, {
  href: repo.html_url,
  target: '_blank',
  text: repo.name,
});
```

Reply:
ahh I get it now :) another function for the "toolbox" ^^ thank you for showing me the correct way to do this, I have included an if statement here for repositories without a description but I wonder if there is a more elegant way of handling it since it kinda doubles the code 🤷‍♀